signingscript/test/integration/test_autograph.py

```diff
@@ -182,37 +197,45 @@ def _assert_apk_signature(keystore_path, apk_path, certificate_alias):
     command = subprocess.run(
         cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True
     )
-    assert command.returncode == 0
+    return command.returncode == 0
 
-    files_in_meta_inf = [
-        line for line in command.stdout.split('\n')
-        if 'META-INF/' in line
-    ]
-    assert len(files_in_meta_inf) == 3
-    assert any('META-INF/MANIFEST.MF' in line for line in files_in_meta_inf)
-    assert any('META-INF/SIGNATURE.SF' in line for line in files_in_meta_inf)
-    assert any('META-INF/SIGNATURE.RSA' in line for line in files_in_meta_inf)
+
+def _extract_compress_type_per_filename(path):
+    with zipfile.ZipFile(path) as zip:
+        return {
+            zip_info.filename: zip_info.compress_type
+            for zip_info in zip.infolist()
+        }
 
 
 @pytest.mark.asyncio
 @skip_when_no_autograph_server
-@pytest.mark.parametrize('file_name', ('app.apk', 'app_preexisting_signature.apk',)
-)
-async def test_integration_autograph_apk(context, tmpdir, file_name):
+async def test_integration_autograph_focus(context, tmpdir):
+    file_name = 'app.apk'
     original_file_path = os.path.join(TEST_DATA_DIR, file_name)
     copied_file_folder = os.path.join(context.config['work_dir'], 'cot', 'upstream-task-id1')
     makedirs(copied_file_folder)
     shutil.copy(original_file_path, copied_file_folder)
 
+    zip_infos_before_signature = _extract_compress_type_per_filename(os.path.join(copied_file_folder, file_name))
+
     context.config['signing_server_config'] = _write_server_config(tmpdir)
     context.task = _craft_task([file_name], signing_format='autograph_focus')
 
     keystore_path = os.path.join(tmpdir, 'keystore')
     certificate_path = os.path.join(TEST_DATA_DIR, 'autograph_apk.pub')
-    certificate_alias = 'autograph_apk'
+    certificate_alias = 'autograph_focus'
     _instanciate_keystore(keystore_path, certificate_path, certificate_alias)
 
     await async_main(context)
 
     signed_path = os.path.join(tmpdir, 'artifact', file_name)
```

Comment:
Unclear to me who's generating this signed bit but I'll dig after reviewing to not block this PR.

Reply:
Ah, that's smooth, nice! 👍